imitation/scripts/train.py

```diff
@@ -38,130 +43,149 @@ def train_and_plot(policy_dir, env='CartPole-v1',
     - Plot the performance of the generator policy versus the performance of
       a random policy.
 
-  Params:
-    - ...
+  Args:
+      policy_dir: Path to a directory that holds pickled expert policies.
+      env: The environment to train in, by default 'CartPole-v1'. This
+          argument is ignored if the `trainer` is manually provided via the
+          `trainer` argument.
+      n_epochs: The number of epochs to train. Each epoch consists of
+          `n_disc_steps_per_epoch` discriminator steps followed by
+          `n_gen_steps_per_epoch` generator steps.
+      n_epochs_per_plot: An optional number, greater than or equal to 1. The
+          (possibly fractional)
+          number of epochs between each plot. The first plot is at epoch 0,
+          right after the first discrim and generator steps.
+
+          If `n_epochs_per_plot is None`, then don't make any plots.
+      n_disc_steps_per_epoch: The number of discriminator update steps during
+          every training epoch.
+      n_gen_plot_episodes: The number of generator update steps during every
+          generator epoch.
+      n_episodes_per_reward_data: The number of episodes to average over when
+          calculating the average episode reward of a policy.
+      interactive: Figures are always saved to "output/*.png". If `interactive`
+        is True, then also show plots as they are created.
+      trainer: If this is provided, then start training on this AIRLTrainer
+        instead of initializing a new one using `airl_trainer_kwargs`. Also,
+        ignore the `env` argument.
   """
+  assert n_epochs_per_plot is None or n_epochs_per_plot >= 1
   if trainer is None:
+    assert env is not None
     trainer = init_trainer(env, policy_dir=policy_dir, **trainer_kwargs)
-  if trainer_hook_fn:
-    trainer_hook_fn(trainer)
+  env = trainer.env
```

Comment:
Check that `env` is None? If we must have two ways of specifying the same thing, at least make sure users are certain which one they're using.

Reply:
ended up just removing the optional `trainer` parameter